src/components/Asset/AssetActions/Download.tsx

```diff
@@ -73,7 +73,11 @@ export default function Download({
     setValidOrderTx(asset?.accessDetails?.validOrderTx)
     // get full price and fees
     async function init() {
-      if (asset?.accessDetails?.addressOrId === ZERO_ADDRESS) return
+      if (
+        asset?.accessDetails?.addressOrId === ZERO_ADDRESS ||
+        asset?.accessDetails.type === 'free'
```

Comment:
if there's `asset?.accessDetails?` everywhere else, the `?` should be set here too